packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -77,20 +79,23 @@ const getAccountInfo = async (request: Request<MessageTypes.GetAccountInfo>) =>
 
     const transactionsPage = await fetchTransactionPage(request);
 
-    // TODO(vl): parse transaction page
+    const transactions = transactionsPage
+        .map(tx => solanaUtils.transformTransaction(tx, payload.descriptor))
+        .filter((tx): tx is Transaction => !!tx);
 
     const account: AccountInfo = {
         descriptor: payload.descriptor,
         balance: accountInfo.lamports.toString(), // TODO(vl): check if this should also include staking balances
         availableBalance: accountInfo.lamports.toString(), // TODO(vl): revisit to make sure that what getAccountInfo returns is actually available balance
-        empty: accountInfo.lamports === 0, // TODO(vl): this is not correct, it should depend on the length of transaction history
-        // TODO(vl): transaction history
+        empty: !!transactions.length,
```

Comment:
This is wrong, it should the opposite, sorry, fixed in https://github.com/trezor/trezor-suite/pull/9498/commits/d523c1e0a6f6212b1d2b95ba29da834d0d03ceaa